src/devtools/client/webconsole/components/Search/useConsoleSearch.test.tsx

```diff
@@ -0,0 +1,166 @@
+import React from "react";
+import { act } from "react-dom/test-utils";
+import {
+  createTestStore,
+  filterCommonTestWarnings,
+  render as testUtilsRender,
+} from "test/testUtils";
+import useConsoleSearch from "./useConsoleSearch";
+import { Message } from "../../reducers/messages";
+
+import type { Actions, State } from "./useConsoleSearch";
+import type { UIStore } from "ui/actions";
+import { ValueFront } from "protocol/thread";
+import { UIState } from "ui/state";
+
+describe("useConsoleSearch", () => {
+  filterCommonTestWarnings();
+
+  let lastCommitedSearchActions: Actions | null = null;
+  let lastCommitedSearchState: State | null = null;
+  let messageIdCounter: number = 0;
+  let TestComponent: any;
+
+  beforeEach(() => {
+    lastCommitedSearchActions = null;
+    lastCommitedSearchState = null;
+    messageIdCounter = 0;
+
+    TestComponent = () => {
+      const [state, actions] = useConsoleSearch();
+      lastCommitedSearchActions = actions;
+      lastCommitedSearchState = state;
+
+      return null;
+    };
+  });
+
+  // TODO [bvaughn] This is pretty gross; can we collocate this (closer to the Redux code if nothing else)?
```

Comment:
cc @markerikson Any thoughts about this kind of test?

Reply:
Heh, my brain's fried enough from being social at the conf today I'm not sure I can come up with specific feedback atm :)

Totally agreed that we're going to want a better testing environment going forward. There's just so much complexity around the store setup and data loading in the codebase atm.